src/components/Calendar/Event/EditEvent.js

```diff
@@ -0,0 +1,286 @@
+                    />
+                </LocalizationProvider>
+            </div>
+        </FormControl>
+    );
+
+    function handleReccurenceChange(e) {
+        setEventData({ ...eventData, reccurence: e.target.value})
+    }
+
+    function handleEvent() {
+        // TODO:  validate user inputs if have time
+        navigate('/calendar');
+        axios.post(`${process.env.REACT_APP_BASE_URL}events/update`, eventData)
+            .then(() => {
+                navigate('/calendar');
+            })
+            .catch(err => {
+                seteventError({ ...eventError, message: "Error connecting to database. " + err });
+                seteventError({ ...eventError, hasError: true });
+            });
+    }
+
+    function handleCancel() {
+        navigate('/calendar')
+    }
+
+   
+
+ function handleEventHeaderChange(e) {
+        setEventData( {...eventData, eventHeader: e.target.value })
+        console.log(eventData)
+    }
+
+
+    function handleDescriptionChange(e) {
+        setEventData({ ...eventData, description: e.target.value })
+        console.log(eventData)
+    }
+
+    function handleStartDateChange(e) {
+        setEventData({ ...eventData, startDate: e.$d })
+    }
+
+    function handleStartTimeChange(e) {
+        setEventData({ ...eventData, startTime: e.target.value})
+    }
+
+    function handleEndTimeChange(e) {
+        setEventData({ ...eventData, endTime: e.target.value})
+    }
+
+    const PageError = eventError.hasError ? (
+        <Typography align="center" color="#DA3A16">
+            {eventError.message}
+        </Typography>
+    ) : null;
+
+    function handleIsReccurentChange() {
+        setIsRecurrent((prev) =>  !prev);
+    }
+
+    const buttons = (
+        <React.Fragment>
+            <div style={{ paddingTop: '20px'}}>
+                <PrimaryButton2 width='305px' colour={'#912338'} content="Update" onClick={()=>{handleEvent()}} />
+            </div>
+            <div style={{ paddingTop: '20px'}}>
+                <SecondaryButton2 width='305px' content="Cancel" onClick={handleCancel} />
+            </div>
+        </React.Fragment>
+    );
+
+    const addEventForm = (
+        <React.Fragment>
+            <form style={{ paddingLeft: '10px', paddingRight: '10px' }}>
+                <div style={{ paddingTop: '10px', paddingBottom: '10px' }}>{PageError}</div>
+                <Typography align='center' style={{ fontFamily: 'Roboto', fontSize: '30px', fontWeight: 'bold' }}>
+                    Edit Event
+                </Typography>
+                <div align='center' style={{ paddingTop: '16px', paddingBottom: '20px' }}>
+
+                    {/** Event name 9ffc10ba-0149-4468-939a-836fa63d8aa6 
+                     * e090b0c8-91d4-4cd6-9ed5-858f25048aa1
+                    */}
+                    <div style={{ paddingTop: '10px', paddingBottom: '10px' }}>
+                        <TextField
```

Comment:
for all sections who are copy pasted from the create events, might want to create custom mui components for them, so we have less duplicate code. Ex : call <EventHeaderInput> instead of line 164-175.